### PR TITLE
Add `--single-process` argument to workbench commandline arguments

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

Add a flag to workbench to avoid starting 2 processes. See the first commit message for more details.

**To test:**

* For **PyCharm** users, follow the updated instructions by adding the new `--single-process` argument
  to the `Parameters` box, setting a breakpoint in Python and verifying it is hit
* For **CLion** users, in the workbench configuration editor add `--single-process` to the Program arguments and setting a breakpoint in C++ and verifying it is hit
* For **Visual Studio** users, run the workbench debug target. setting a breakpoint in C++ and verifying it is hit.

*There is no associated issue.*

*This does not require release notes* because **it is an internal change.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by martyngigg** [comment]
@gemmaguest @sf1919 @DannyHindson I've just made a change to simplify debugging workbench so that it no longer requires either attaching or remote debugging. 

If you get a second could check this works for you in CLion (@gemmaguest), PyCharm (@sf1919), Visual Studio (@DannyHindson).

**Review by DannyHindson — CHANGES_REQUESTED** [review]

File: qt/applications/workbench/CMakeLists.txt [hunk 1]
```diff
@@ -12,12 +12,15 @@ if(APPLE OR WIN32)
     INSTALL_LIB_DIRS "${WORKBENCH_SITE_PACKAGES}"
   )
   # Allow VS to start workbench for debugging
+  # cmake-format: off
   set_target_properties(
     workbench
     PROPERTIES VS_DEBUGGER_COMMAND $<$<BOOL:${WIN32}>:${Python_EXECUTABLE}>
-               VS_DEBUGGER_COMMAND_ARGUMENTS $<$<BOOL:${WIN32}>:${MSVC_BIN_DIR}/workbench-script.pyw>
-               VS_DEBUGGER_ENVIRONMENT $<$<BOOL:${WIN32}>:${MSVC_IDE_ENV}>
+               VS_DEBUGGER_COMMAND_ARGUMENTS $<$<BOOL:${WIN32}>:"${MSVC_BIN_DIR}/workbench-script.pyw --single-process">
```
[the remark this answers is not in the document]
Comment (DannyHindson) [reply qt/applications/workbench/CMakeLists.txt]: I've just retested this and running the workbench target works fine now in Visual Studio. I set a breakpoint in an algorithm exec() and it was hit. Also if I start workbench using the new --single-process parameter from within PyCharm and attach Visual Studio to the process with the intention of debugging the C++ code, I now only see a single python process:

image

I also tested the behaviour in relation to hitting python breakpoints in PyCharm Professional on Windows. For some reason PyCharm already hit breakpoints for me in the workbench code without using the remote debugger feature before this change. I did have to use the remote debugger to hit breakpoints in python algorithm code. This is all still the case with the new parameter so it's not broken anything even if there's no improvement there.

**Comment by gemmaguest** [comment]
Works well on CLion on Ubuntu - simple to set up, attaches and hits breakpoints fine :+1:

**Comment by martyngigg** [comment]
Would we be happy for this to be merged now?

**Comment by DannyHindson** [comment]
> Would we be happy for this to be merged now?

Yes from me. I'll approve it now

**Review by DannyHindson — APPROVED** [review]

**Pull request merged**